tests/test_utils.py

```diff
@@ -20,3 +42,76 @@ def test_get_s2_evalscript(all_bands: bool, reflectance: bool) -> None:
     assert input_units_str in evalscript
     output_sample_type_str = 'sampleType: "FLOAT32"' if reflectance else 'sampleType: "UINT16"'
     assert output_sample_type_str in evalscript
+
+
+@pytest.mark.parametrize(
+    "test_input, expected",
+    [
+        ({"bbox": BBOX1, "time_interval": (dt.datetime(2021, 1, 1), dt.datetime(2021, 1, 1))}, []),
+        (
+            {"bbox": BBOX1, "time_interval": (dt.datetime(2021, 1, 1), dt.datetime(2021, 1, 10))},
+            [
+                dt.datetime(2021, 1, 4, 16, 39, 29),
+                dt.datetime(2021, 1, 4, 16, 39, 44),
+                dt.datetime(2021, 1, 9, 16, 39, 29),
+                dt.datetime(2021, 1, 9, 16, 39, 43),
+            ],
+        ),
+        (
+            {
+                "bbox": BBOX1,
+                "time_interval": (dt.datetime(2021, 1, 1), dt.datetime(2021, 1, 10)),
+                "time_difference": dt.timedelta(seconds=600),
+            },
+            [dt.datetime(2021, 1, 4, 16, 39, 29), dt.datetime(2021, 1, 9, 16, 39, 29)],
+        ),
+    ],
+)
+@pytest.mark.sh_integration
+def test_get_timestamps(test_input: dict, config: SHConfig, expected: List[dt.datetime]) -> None:
+    timestamps = get_timestamps(**test_input, config=config)
+    assert [timestamp.replace(tzinfo=None) for timestamp in timestamps] == expected
+
+
+@pytest.mark.parametrize(
+    "test_input, expected_shape",
+    [
+        ({"bbox": BBOX1, "timestamps": [dt.datetime(2021, 1, 4)], "size": (60, 81)}, (1, 81, 60, 13)),
+        (
+            {
+                "bbox": BBOX1,
+                "timestamps": [dt.datetime(2021, 1, 4, 16, 39, 29), dt.datetime(2021, 1, 9, 16, 39, 29)],
+                "size": (60, 81),
+                "all_bands": True,
+            },
+            (2, 81, 60, 13),
+        ),
+        (
+            {
+                "bbox": BBOX1,
+                "timestamps": [dt.datetime(2021, 1, 4, 16, 39, 29), dt.datetime(2021, 1, 9, 16, 39, 29)],
+                "size": (60, 81),
+                "all_bands": False,
+            },
+            (2, 81, 60, 10),
+        ),
+        (
+            {
+                "bbox": BBOX2,
+                "timestamps": [dt.datetime(2016, 7, 18, 7, 14, 4)],
+                "resolution": (100, 100),
```

Comment:
cool, could we perhaps use resolution 500 to reduce the amount of data? (unless sentinelhub complains)

Reply:
it dose
`Server response: "{"status": 400, "reason": "Bad Request", "message": "Your request of 498.62 meters per pixel exceeds the limit 200.00 meters per pixel of the collection S2L1C. Please revise the resolution (or corresponding width/height) to make sure it is in supported range.", "code": "RENDERER_EXCEPTION"}` 
but could be 200

Reply:
could be 200 , judging from the error message :)
perhaps shrink BBOX2 a bit as well?

Reply:
done :)